Tribler/community/tunnel/hidden_community.py

```diff
@@ -312,26 +398,34 @@ def on_key_request(self, messages):
 
     def check_key_response(self, messages):
         for message in messages:
-            if not message.source.startswith(u"circuit_"):
-                yield DropMessage(message, "must be received from a circuit")
-                continue
-
+            self._logger.debug("Check key response")
             request = self.request_cache.get(u"key-request", message.payload.identifier)
             if not request:
                 yield DropMessage(message, "invalid key-response identifier")
                 continue
-
             yield message
 
     def on_key_response(self, messages):
         for message in messages:
-            self._logger.debug("On key response: received keys")
-            cache = self.request_cache.pop(u"key-request", message.payload.identifier)
-            self.create_e2e(cache.circuit, cache.sock_addr, cache.info_hash, message.payload.public_key)
```

Comment:
I'm thinking now, that it's caused by replying to the exitsocket of the downloader (Dean) with the wrong socket